src/rgw/s3select.h

```diff
@@ -723,37 +752,28 @@ class csv_object : public base_s3object
     std::string m_error_description;
     char *m_stream;
     char *m_end_stream;
-    std::vector<std::string_view> m_row_tokens{128};
+    std::vector<char*> m_row_tokens{128};
```

Comment:
can this overflow?

Reply:
no, per token its test against vector capacity (main parse loop)
tests , to validate that will be added